libs/dh/shared/feature-authorization/src/lib/scope.service.ts

```diff
@@ -16,18 +16,105 @@
  */
 
 import { Inject, Injectable } from '@angular/core';
+import { MsalBroadcastService, MsalService } from '@azure/msal-angular';
+import { EventMessage, EventType } from '@azure/msal-browser';
 import {
   DhB2CEnvironment,
   dhB2CEnvironmentToken,
 } from '@energinet-datahub/dh/shared/environments';
+import { DhFeatureFlagsService } from '@energinet-datahub/dh/shared/feature-flags';
+import { filter } from 'rxjs';
+import { ScopeStorage, scopeStorageToken } from './scope-storage.service';
+
+export const scopesKey = 'actor-scopes';
+export const scopeKey = 'active-actor-scope';
 
 @Injectable({ providedIn: 'root' })
 export class ScopeService {
   constructor(
-    @Inject(dhB2CEnvironmentToken) private config: DhB2CEnvironment
-  ) {}
+    @Inject(dhB2CEnvironmentToken) private config: DhB2CEnvironment,
+    private msalBroadcastService: MsalBroadcastService,
+    private authService: MsalService,
+    private featureFlagService: DhFeatureFlagsService,
+    @Inject(scopeStorageToken) private scopeStorage: ScopeStorage
+  ) {
+    this.msalBroadcastService.msalSubject$
+      .pipe(
+        filter((msg: EventMessage) => msg.eventType === EventType.LOGIN_SUCCESS)
+      )
+      .subscribe(() => {
+        const account = this.getAccount();
+
+        if (!account) {
+          this.clearScopeRelatedItems();
+          return;
+        }
+
+        const userId = account.localAccountId;
+        const actors =
+          account.idTokenClaims &&
+          (account.idTokenClaims['extn.actors'] as string[]);
```

Comment:
suggestion: consider moving `extn.actors` to a variable as well. It's used in the spec file too.

Reply:
fixed